### PR TITLE
Implemented Responsiveness, Mapping Data to PieChart and Total

### PR DESCRIPTION
**Pull request by Mihaidaniel22**

### CODE REVIEW

File: src/components/homepage/CustomPieChart.jsx [hunk 1]
```diff
@@ -1,48 +1,93 @@
...
 export const CustomPieChart = () => {
   const [selectedDate, setSelectedDate] = useState(new Date());
+  const [userData, setUserData] = useState([]);
 
   const handleDateChange = (newDate) => {
     setSelectedDate(newDate);
   };
+
+  useEffect(() => {
+    const fetchData = async () => {
```
Comment (en-teodora-velicu) [inline src/components/homepage/CustomPieChart.jsx:18]: This should be in a service

File: src/components/homepage/CustomPieChart.jsx [hunk 2]
```diff
@@ -1,48 +1,93 @@
...
+  }, []);
+
+  const mapDataToChart = (dataToMap) => {
+    const dataObject = [];
+
+    for (const [key, value] of Object.entries(dataToMap)) {
+
+      const numericValue = parseFloat(value);
+      let categoryColor = '';
+      if (key === "Food & Restaurants")
```
Comment (en-teodora-velicu) [inline src/components/homepage/CustomPieChart.jsx:43]: This is better with a switch case and also extract it in a new function

File: src/components/homepage/DateNavigation.jsx [hunk 3]
```diff
@@ -72,7 +77,11 @@ export const DateNavigation = ({ selectedDate, onDateChange }) => {
           }}
         >
           <LocalizationProvider dateAdapter={AdapterDateFns}>
-            <DateCalendar views={["day", "month", "year"]} />
+            <DateCalendar
+              views={["day", "month", "year"]}
```
Comment (en-teodora-velicu) [inline src/components/homepage/DateNavigation.jsx:81]: this is an array of constants. move outside the component

File: src/components/homepage/TotalExpenses.jsx [hunk 4]
```diff
@@ -1,13 +1,34 @@
...
 import { Box, IconButton } from "@mui/material";
 import AddIcon from "@mui/icons-material/Add";
 import { TOTAL_EXPENSES, NAV_ITEMS } from "../../constants/constants";
+import { getTotal } from "../../services/total-service";
 
 export const TotalExpenses = () => {
+  const [total, setTotal] = useState([]);
+
+  useEffect(() => {
+    const fetchData = async () => {
```
Comment (en-teodora-velicu) [inline src/components/homepage/TotalExpenses.jsx:12]: Extract in a service for expenses

File: src/services/categories-service.jsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    console.error("Error fetching categories:", error);
+    throw error;
+  }
+};
+
+export const getDataForChart = async () => {
+  // export const getDataForChart = async (userId, startDate, endDate, code) => {
+
+  try {
+    // const response = await apiClient().get("http://localhost:8080/expense/category", {
```
Comment (en-teodora-velicu) [inline src/services/categories-service.jsx:20]: remove commented code

File: src/services/total-service.jsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import apiClient from "./ApiClient-service";
+
+export const getTotal = async () => {
```
Comment (en-teodora-velicu) [inline src/services/total-service.jsx:3]: total service? you should create you services having in mind a broader scope. total what? total expenses? categories what? expense categories chart? you can have a service expenses.js(for all info regarding costs) reports.js(the endpoints used to create the reports page)

**Review by en-teodora-velicu — CHANGES_REQUESTED** [review]

**Review by en-teodora-velicu — APPROVED** [review]

**Review by CristiStroescu — APPROVED** [review]

**Review by Alexandra18042003 — APPROVED** [review]

**Pull request merged**